openpaas-james/guice/encrypted-mailbox/pom.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.linagora.openpaas</groupId>
+        <artifactId>openpaas-james</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>encrypted-mailbox-guice</artifactId>
+    <packaging>jar</packaging>
+
+    <name>Apache James :: Server :: ElasticSearch - guice modules - v6</name>
+    <description>Guice injections for the EncryptedMailbox</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.james</groupId>
+            <artifactId>james-server-guice-configuration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>mailbox-encrypted-memory</artifactId>
```

Comment:
After thinking again, shouldn't we depend on `mailbox-encrypted-api` here instead ? I'm reaching a similar issue when trying to add the KeystoreModule for distributed james... We might have to do a little refactoring here I think...

Reply:
mailbox-encrypted-memory could be named `mailbox-encrypted-implem` IMO as it contains the EncryptedMailboxManager.

Would it suite you?

Reply:
I don't think a renaming is the solution here... that you call it `mailbox-encrypted-implem` or `mailbox-encrypted-memory` it is the same. With this you inject the `InMemoryKeystoreManager` class in distributed versions as well, which is unecessary.

I think it is the `EncryptedMailboxManager` and `EncryptedMessageManager` that should be moved to `mailbox-encrypted-api` instead. (or implem... changing the name doesn't change the dep itself) It makes no sense to inject memory related modules into distributed, and those 2 classes are generic APIs IMO

Reply:
That could be done in an other PR though... but it is the 2nd time I catch such a thing this sprint. We need to be more careful (I include myself as this slipped my review, so I'm as faulty^^)